src/idsk/components/customer-surveys/customer-surveys.js

```diff
@@ -0,0 +1,212 @@
+
+    if ($radioButtonWork) {
+        var $self = this;
+        // Handle $radioButton click events
+        $radioButtonWork.addEventListener('click', $self.handleRadioButtonWorkClick.bind($self));
+    }
+
+    if ($radioButtonPrivate) {
+        var $self = this;
+        // Handle $radioButton click events
+        $radioButtonPrivate.addEventListener('click', $self.handleRadioButtonPrivateClick.bind($self));
+    }
+
+    if ($nextButton) {
+        $nextButton.addEventListener('click', this.handleNextButtonClick.bind(this));
+    }
+
+    if ($previousButton) {
+        $previousButton.addEventListener('click', this.handlePreviousButtonClick.bind(this));
+    }
+
+    if ($textAreaFirst) {
+        $module.textAreaMap.set('first', 1);
+        $textAreaFirst.addEventListener('input', this.handleStatusOfCharacterCountButton.bind(this, 'first'));
+    }
+
+    if ($textAreaSecond) {
+        $module.textAreaMap.set('second', 2);
+        $textAreaSecond.addEventListener('input', this.handleStatusOfCharacterCountButton.bind(this, 'second'));
+    }
+
+    if ($textAreaThird) {
+        $module.textAreaMap.set('third', 4);
+        $textAreaThird.addEventListener('input', this.handleStatusOfCharacterCountButton.bind(this, 'third'));
+    }
+
+    if ($textAreaFourth) {
+        $module.textAreaMap.set('fourth', 5);
+        $textAreaFourth.addEventListener('input', this.handleStatusOfCharacterCountButton.bind(this, 'fourth'));
+    }
+};
+
+CustomerSurveys.prototype.handleStatusOfCharacterCountButton = function (e, $name) {
+    var $module = this.$module;
+    var $textAreaCharacterCount = $module.querySelector('#' + $name.srcElement.id);
+    var $remainingCharacterCountMessage = $module.querySelector('#' + $name.srcElement.id + '-info');
+    var $submitButton = $module.querySelector('#idsk-customer-surveys__send-button');
+
+    setTimeout(function () {
+        if ($textAreaCharacterCount.classList.contains('govuk-textarea--error') || $remainingCharacterCountMessage.classList.contains('govuk-error-message')) {
+            $submitButton.disabled = true;
+        } else {
+            $submitButton.disabled = false;
+            // changing value of global variable for disabling button, in case of walk through steps and comming back to this textarea.
+            $module.sendButtonDisabled[$module.textAreaMap.get($name.srcElement.id)] = false;
+        }
+    }, 300);
+}
+
+CustomerSurveys.prototype.handleCounterOfSubtitles = function ($counter) {
+    var $subtitles = this.$module.querySelectorAll('.idsk-customer-surveys--subtitle');
+    var i;
+
+    // remove previous indexing, cause amount of steps could change
+    for (i = 0; i < $counter; i++) {
+        $subtitles[i].textContent = $subtitles[i].textContent.substring(3);
+    }
+
+    // adding new indexing
+    for (i = 0; i < $counter; i++) {
+        $subtitles[i].innerHTML = (i + 1) + '. ' + $subtitles[i].textContent;
+    };
```

Comment:
zbytocne 2 cykly.. vies to srpavit v jednom ... idealne cez forEach

Reply:
suhlasim dal som to do jedneho ale foreach nemoze kvoli tomu ze posielam pocet opakovani v parametri funkcie